x-pack/platform/plugins/shared/fleet/server/collectors/agentless_collectors.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ElasticsearchClient, SavedObjectsClient } from '@kbn/core/server';
+
+import { agentPolicyService } from '../services';
+
+import type { NewPackagePolicy } from '../types';
+
+import { getAgentUsage } from './agent_collectors';
+import type { AgentUsage } from './agent_collectors';
+
+export const getAgenlessUsage = async (
+  soClient?: SavedObjectsClient,
+  esClient?: ElasticsearchClient
+): Promise<AgentUsage> => {
+  if (!soClient) {
+    return {
+      total_enrolled: 0,
+      healthy: 0,
+      unhealthy: 0,
+      offline: 0,
+      inactive: 0,
+      unenrolled: 0,
+      total_all_statuses: 0,
+      updating: 0,
+    };
+  }
+
+  const agentPolicies = await agentPolicyService.list(soClient, {
```

Comment:
It seems there is a `supports_agentless` property on the agent policy too, we can probably simplify this and do not fetch the package policies, similar to how it's done [here](https://github.com/elastic/kibana/blob/main/x-pack/platform/plugins/shared/fleet/server/services/outputs/helpers.ts#L23)

Reply:
Nice ! Done, just why the index(`LEGACY_AGENT_POLICY_SAVED_OBJECT_TYPE`) is legacy? should I use different / additional one?

Reply:
We currently have two saved object type for agent policies, one that is space aware, one that is not, using any of them in the kuery will work as the agentPolicy service will rewrite the kuery if space awareness is enabled for fleet or not.